### PR TITLE
CLI improvements

### PR DESCRIPTION
**Pull request by SebastianSchmidl**
add important CLI capabilities for the final assignment

### CODE REVIEW

**Comment by SebastianSchmidl** [comment]
please review based on Assignment 8 task description

File: cse/SearchEngine.py [hunk 1]
```diff
@@ -99,33 +94,40 @@ def index(self):
...
 
 
-    def search(self, query, topK=10):
+    def search(self, query, idsOnly=False, topK=10):
         if not self.__indexLoaded:
             print("Index was not loaded!")
             return []
 
         results = []
+        print()
```
Comment (Benedikt1992) [inline cse/SearchEngine.py:116]: ?

File: searchengine.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    parser = argparse.ArgumentParser(description='Search Engine for comments on news websites')
+    parser.add_argument("query", help="a txt file with one boolean, keyword, phrase, ReplyTo, or Index query per line")
+    parser.add_argument("--topN", help="the maximum number of search hits to be printed", type=int)
+    parser.add_argument("--printIdsOnly", help="print only commentIds and not ids and their corresponding comments", action="store_true")
+    args = parser.parse_args()
+
+    # load queries
+    shouldBuildIndex = False
+    commentsFile, authorFile, articleFile = COMMENTSFILE, AUTHORFILE, ARTICLEFILE
+    if args.query.startswith("Index:"):
```
Comment (Benedikt1992) [inline searchengine.py:57]: This will be the first line in the queries textfile

File: searchengine.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        shouldBuildIndex = True
+        commentsFile, authorFile, articleFile = checkIndexFilenames(args.query)
+        
+    else:
+        queries = read_queries(args.query)
+        if queries and queries[0].startswith("Index:"):
```
Comment (Benedikt1992) [inline searchengine.py:63]: nevermind...

File: cse/SearchEngine.py (hunk 1 above)
Comment (SebastianSchmidl) [reply cse/SearchEngine.py:116]: prints a newline to separate the output of multiple calls to `search()`

File: searchengine.py (hunk 2 above)
Comment (SebastianSchmidl) [reply searchengine.py:57]: That's the way I understood the assignment the first time, as well. Therefore, I kept the possibility to have both (the first line in the query-file and as the name of the file)

**Review by Benedikt1992 — APPROVED** [review]

**Pull request merged**